parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -164,21 +164,7 @@ write_files:
   permissions: "0644"
   owner: root
   content: |
-    {
-      "live-restore": true,
-      "log-driver": "json-file",
-      "log-opts":  {
-         "max-size": "50m",
-         "max-file": "5"
-      }{{if IsNSeriesSKU .}}
-      ,"default-runtime": "nvidia",
-      "runtimes": {
-         "nvidia": {
-             "path": "/usr/bin/nvidia-container-runtime",
-             "runtimeArgs": []
-        }
-      }{{end}}
```

Comment:
follow this pattern, we can have something like below, which seems much more easier to read, can we do that and save the code, I really like to have consistent and simple code 
      {{if HasDataDir.}}	
      ,"root": "{{ GetDataDir }}"
      {{end}}

Reply:
I understand your original intention in aks-e code for testability, but since in this repo we haven't added test for all the templates, I think we will have to figure that out for all of the template test.  (For now, I'm having template  testing in aks-rp https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=%2Fresourceprovider%2Fserver%2Fmicrosoft.com%2Fcommon%2Fagentbaker%2Ftestdata)